pinot-common/src/main/java/org/apache/pinot/common/utils/config/TableConfigUtils.java

```diff
@@ -235,34 +233,56 @@ public static void convertFromLegacyTableConfig(TableConfig tableConfig) {
         (ingestionConfig != null) ? ingestionConfig.getBatchIngestionConfig() : null;
 
     SegmentsValidationAndRetentionConfig validationConfig = tableConfig.getValidationConfig();
+    String segmentPushType = validationConfig.getSegmentPushType();
+    String segmentPushFrequency = validationConfig.getSegmentPushFrequency();
+
     if (batchIngestionConfig == null) {
-      batchIngestionConfig = new BatchIngestionConfig(null, validationConfig.getSegmentPushType(),
-          validationConfig.getSegmentPushFrequency());
+      // Only create the config if any of the deprecated config is not null.
+      if (segmentPushType != null || segmentPushFrequency != null) {
+        batchIngestionConfig = new BatchIngestionConfig(null, segmentPushType, segmentPushFrequency);
+      }
     } else {
       // This should not happen typically, but since we are in repair mode, might as well cover this corner case.
       if (batchIngestionConfig.getSegmentIngestionType() == null) {
-        batchIngestionConfig.setSegmentIngestionType(validationConfig.getSegmentPushType());
+        batchIngestionConfig.setSegmentIngestionType(segmentPushType);
       }
 
       if (batchIngestionConfig.getSegmentIngestionFrequency() == null) {
-        batchIngestionConfig.setSegmentIngestionFrequency(validationConfig.getSegmentPushFrequency());
+        batchIngestionConfig.setSegmentIngestionFrequency(segmentPushFrequency);
       }
     }
 
     StreamIngestionConfig streamIngestionConfig =
         (ingestionConfig != null) ? ingestionConfig.getStreamIngestionConfig() : null;
+    IndexingConfig indexingConfig = tableConfig.getIndexingConfig();
+
     if (streamIngestionConfig == null) {
-      streamIngestionConfig =
-          new StreamIngestionConfig(Collections.singletonList(tableConfig.getIndexingConfig().getStreamConfigs()));
+      Map<String, String> streamConfigs = indexingConfig.getStreamConfigs();
+
+      // Only set the new config if the deprecated one is set.
+      if (streamConfigs != null && !streamConfigs.isEmpty()) {
+        streamIngestionConfig = new StreamIngestionConfig(Collections.singletonList(streamConfigs));
+      }
     }
 
     if (ingestionConfig == null) {
       ingestionConfig = new IngestionConfig(batchIngestionConfig, streamIngestionConfig, null, null, null);
```

Comment:
Set `ingestionConfig` only when one of `batchIngestionConfig` and `streamIngestionConfig` is not `null`